backend/LexBoxApi/Services/HgService.cs

```diff
@@ -75,6 +75,39 @@ private void InitRepoAt(string code)
         );
     }
 
+    public async Task PrepareEmptyRepo(string code, string tempRepoSuffix)
+    {
+        var tempRepoName = $"{code}__{tempRepoSuffix}";
+        await Task.Run(() =>
+        {
+            var deletedRepoPath = Path.Combine(_options.Value.RepoPath, DELETED_REPO_FOLDER);
+            var directory = Directory.CreateDirectory(deletedRepoPath);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                directory.UnixFileMode = Permissions;
+            var dest = Directory.CreateDirectory(Path.Combine(directory.FullName, tempRepoName));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                dest.UnixFileMode = Permissions;
+            FileUtils.CopyFilesRecursively(
+                new DirectoryInfo("Services/HgEmptyRepo"),
+                dest,
+                Permissions
+            );
```

Comment:
This is mostly a duplicate of `InitRepoAt`. How about we make `InitRepoAt` take a path and rename the current one to `InitRepoForCode` or something and then share more code.
Also, settings the file permissions on the destination is covered by `CopyFilesRecursively` so it's redundant.
```suggestion
            var dest = Directory.CreateDirectory(Path.Combine(directory.FullName, tempRepoName));
            FileUtils.CopyFilesRecursively(
                new DirectoryInfo("Services/HgEmptyRepo"),
                dest,
                Permissions
            );
```

Reply:
Done in commit b2f6ef07.